### PR TITLE
Chi 347 add ability to search documentation

### PR DESCRIPTION
**Pull request by janibolkvadze**
@mattnickles @jllr @dani-cl-madrid @SergioQCL

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build STARTED. See http://jenkins.elasticbox.com/job/chi_pull_request/326/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/326/ for more details.

**Comment by janibolkvadze** [comment]
test this please

**Comment by janibolkvadze** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/330/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/333/ for more details.

File: src/website/assets/scripts/algoliaSearch.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }
+});
+
+search.addWidgets([
+  instantsearch.widgets.searchBox({
+    container: '#searchbox',
+    placeholder: "Search...",
+    queryHook(query, refine) {
+      clearTimeout(timerId);
+      timerId = setTimeout(() => refine(query), 1000);
```
Comment (mattnickles) [inline src/website/assets/scripts/algoliaSearch.js]: @janibolkvadze @jllr @dani-cl-madrid very slow. taking far too long to show and hide results

File: src/website/assets/scripts/algoliaSearch.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        searchResults.style.width = '300px';
+      });
+
+      if(!aisSearchBoxForm.classList.contains("a-inputWrapper")) {
+        aisSearchBoxForm.classList.add("a-inputWrapper");
+        aisSearchBoxForm.classList.add("-icon--left");
+        aisSearchBoxForm.appendChild(icon);
+      }
+
+      if (helper.state.query.length < 3) {
```
Comment (mattnickles) [inline src/website/assets/scripts/algoliaSearch.js]: @janibolkvadze @jllr @dani-cl-madrid can we reduce to 1? Algolia docsearch defaults to 1 and feels much faster

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/342/ for more details.

File: src/website/assets/scripts/algoliaSearch.js (hunk 2 above)
Comment (janibolkvadze) [reply src/website/assets/scripts/algoliaSearch.js]: @mattnickles New PR is implemented using Algolia Docsearch. I think it's a much better option for us. Thanks for the suggestions!

**Comment by jllr** [comment]
@janibolkvadze add flex: 1 0 auto to o-header__title in md viewport

File: src/chi/components/header/header.scss [hunk 3]
```diff
@@ -102,6 +102,70 @@
...
+      .a-input {
+        width: 500px;
+      
+        &:focus {
+          width: 600px;
+        }
+      }
+
+      .algolia-autocomplete .ds-dropdown-menu {
+        min-width: 600px;
```
[the remark this answers is not in the document]
Comment (janibolkvadze) [reply src/chi/components/header/header.scss]: All done, thanks

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/345/ for more details.

File: src/website/assets/styles/_docs-header.scss [hunk 4]
```diff
@@ -66,6 +66,73 @@
             font-size: $text-larger;
           }
         }
+
+        .-chi-search {
+          .a-input {
+            width: 18.75rem;
+            height: 2.5rem!important;
```
Comment (SergioQCL) [inline src/website/assets/styles/_docs-header.scss]:
```suggestion
            height: 2.5rem !important;
```

File: src/website/assets/styles/_docs-header.scss [hunk 5]
```diff
@@ -66,6 +66,73 @@
             font-size: $text-larger;
           }
         }
+
+        .-chi-search {
+          .a-input {
+            width: 18.75rem;
```
Comment (jllr) [inline src/website/assets/styles/_docs-header.scss]: @janibolkvadze please order alphabetically css properties to be consistent

File: src/website/assets/styles/_docs-header.scss [hunk 6]
```diff
@@ -66,6 +66,73 @@
...
+      
+          @include respond-to(sm) {
+            .a-input {
+              width: 12.5rem;
+      
+              &:focus {
+                width: 14rem;
+              }
+      
+              .algolia-autocomplete .ds-dropdown-menu {
```
Comment (jllr) [inline src/website/assets/styles/_docs-header.scss]: @janibolkvadze this class must be out of .a-input class

File: src/website/assets/styles/_docs-header.scss [hunk 7]
```diff
@@ -66,6 +66,73 @@
...
+              &:focus {
+                width: 22rem;
+              }
+            }
+      
+            .algolia-autocomplete .ds-dropdown-menu {
+              min-width: 22rem;
+            }
+          }
+        }
```
Comment (jllr) [inline src/website/assets/styles/_docs-header.scss]: @janibolkvadze please add respond-to(xl) .a-input width: 26rem :focus width : 29rem

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/352/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/353/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/355/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/357/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/358/ for more details.

**Pull request merged**